subworkflows/local/catpack.nf

```diff
@@ -0,0 +1,129 @@
+/*
+ * CAT/BAT/RAT: tools for taxonomic classification of contigs and metagenome-assembled genomes (MAGs) and for taxonomic profiling of metagenomes
+ */
+include { CATPACK_ADDNAMES as CATPACK_ADDNAMES_BINS     } from '../../modules/nf-core/catpack/addnames/main'
+include { CATPACK_ADDNAMES as CATPACK_ADDNAMES_UNBINS   } from '../../modules/nf-core/catpack/addnames/main'
+include { CATPACK_BINS                                  } from '../../modules/nf-core/catpack/bins/main'
+include { CATPACK_CONTIGS as CATPACK_UNBINS             } from '../../modules/nf-core/catpack/contigs/main'
+include { CATPACK_DOWNLOAD                              } from '../../modules/nf-core/catpack/download/main'
+include { CATPACK_PREPARE                               } from '../../modules/nf-core/catpack/prepare/main'
+include { CATPACK_SUMMARISE as CATPACK_SUMMARISE_BINS   } from '../../modules/nf-core/catpack/summarise/main'
+include { CATPACK_SUMMARISE as CATPACK_SUMMARISE_UNBINS } from '../../modules/nf-core/catpack/summarise/main'
+include { UNTAR as CAT_DB_UNTAR                         } from '../../modules/nf-core/untar/main'
+
+
+workflow CATPACK {
+    take:
+    ch_bins   // channel: [ val(meta), [bins] ]
+    ch_unbins // channel: [ val(meta), [unbins] ]
+
+    main:
+    ch_versions = Channel.empty()
+
+    /*
+    ========================================
+     * Database setup
+    ========================================
+     */
+
+    if (params.cat_db) {
```

Comment:
Future-proof thinking: Is it better to pass the DB into this subwf as a channel, rather than reading it direct from the params? In future, it looks like params will only be available in the entry workflow and need to be passed through as arguments.

Reply:
Yeah, it makes more sense to update this across the entire pipeline in a single PR.
Could you open an issue to keep track of this change?